tests/kafkatest/services/verifiable_producer.py

```diff
@@ -114,37 +120,63 @@ def _worker(self, idx, node):
         self.produced_count[idx] = 0
         last_produced_time = time.time()
         prev_msg = None
-        for line in node.account.ssh_capture(cmd):
-            line = line.strip()
-
-            data = self.try_parse_json(line)
-            if data is not None:
-
-                with self.lock:
-                    if data["name"] == "producer_send_error":
-                        data["node"] = idx
-                        self.not_acked_values.append(self.message_validator(data["value"]))
-                        self.produced_count[idx] += 1
-
-                    elif data["name"] == "producer_send_success":
-                        self.acked_values.append(self.message_validator(data["value"]))
-                        self.produced_count[idx] += 1
-
-                        # Log information if there is a large gap between successively acknowledged messages
-                        t = time.time()
-                        time_delta_sec = t - last_produced_time
-                        if time_delta_sec > 2 and prev_msg is not None:
-                            self.logger.debug(
-                                "Time delta between successively acked messages is large: " +
-                                "delta_t_sec: %s, prev_message: %s, current_message: %s" % (str(time_delta_sec), str(prev_msg), str(data)))
-
-                        last_produced_time = t
-                        prev_msg = data
-
-                    elif data["name"] == "shutdown_complete":
-                        if node in self.clean_shutdown_nodes:
-                            raise Exception("Unexpected shutdown event from producer, already shutdown. Producer index: %d" % idx)
-                        self.clean_shutdown_nodes.add(node)
+        node.account.ssh(cmd)
+
+        # Ensure that STDOUT_CAPTURE exists before try to read from it
+        # Note that if max_messages is configured, it's possible for the process to exit before this
+        # wait_until condition is checked
+        start = time.time()
+        wait_until(lambda: node.account.isfile(VerifiableProducer.STDOUT_CAPTURE) and
+                           line_count(node, VerifiableProducer.STDOUT_CAPTURE) > 0,
+                   timeout_sec=10, err_msg="%s: VerifiableProducer took too long to start" % node.account)
+        self.logger.debug("%s: VerifiableProducer took %s seconds to start" % (node.account, time.time() - start))
+
+        with node.account.open(VerifiableProducer.STDOUT_CAPTURE, 'r') as f:
+            while True:
+                line = f.readline()
+                if line == '' and not self.alive(node):
```

Comment:
Is this guaranteed to break properly? It seems like a) the `self.alive(node)` check shouldn't be necessary since the empty string should only happen at the end of the file anyway and b) the empty string check may not be valid given this from the python docs:

> a newline character (\n) is left at the end of the string, and is only omitted on the last line of the file if the file doesn’t end in a newline

I guess we might guarantee this in the Java code? But it seems like it'll be risky to assume it since it seems easy to break this.